test/primitives/object-element-test.js

```diff
@@ -207,6 +207,47 @@ describe('ObjectElement', function() {
       obj.set({ foo: 'bar' });
       expect(obj.get('foo').toValue()).to.equal('bar');
     });
+
+    it('sets the parent to children', function() {
```

Comment:
Parents let you go up and down the tree. Right now, if you get a node from the tree, you cannot know where it is in the tree, only what its children are.